omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -88,41 +98,149 @@ def indented_print(msg, *args, **kwargs):
+            joint_pos = np.array(self.robot.get_joint_positions()[joint_combined_idx])
+            link_poses = self.fk_solver.get_link_poses(joint_pos, arm_links)
+        else:
+            robot_to_copy = self.robot
+
+        # Copy robot meshes
+        for link in robot_to_copy.links.values():
+            for mesh in link.collision_meshes.values():
+                split_path = mesh.prim_path.split("/")
+                link_name = split_path[3]
+                # Do not copy grasping frame (this is necessary for Tiago, but should be cleaned up in the future)
+                if "grasping_frame" in link_name:
+                    continue
+
+                mesh_copy_path = self.robot_copy_path + "/" + link_name
+                mesh_copy_path += f"_{split_path[-1]}" if split_path[-1] != "collisions" else ""
+                mesh_command = CopyPrimCommand(mesh.prim_path, path_to=mesh_copy_path)
+                mesh_command.do()
+                mesh_copy = CollisionGeomPrim(mesh_copy_path, mesh_copy_path)
+                relative_pose = T.relative_pose_transform(*mesh.get_position_orientation(), *link.get_position_orientation())
+                if link_name not in self.robot_meshes_copy.keys():
+                    self.robot_meshes_copy[link_name] = [mesh_copy]
+                    self.robot_meshes_relative_poses[link_name] = [relative_pose]
+                else:
+                    self.robot_meshes_copy[link_name].append(mesh_copy)
+                    self.robot_meshes_relative_poses[link_name].append(relative_pose)
+
+                # Set poses of meshes relative to the robot to construct the robot
+                if self.robot.model_name == "Tiago" and self.mode == "base" and link_name in arm_links:
+                    link_pose = link_poses[link_name]
+                    mesh_copy_pose = T.pose_transform(*link_pose, *relative_pose)
+                    self._set_prim_pose(mesh_copy.prim, mesh_copy_pose)
+                else:
+                    mesh_in_robot = T.relative_pose_transform(*mesh.get_position_orientation(), *robot_to_copy.get_position_orientation())
+                    self._set_prim_pose(mesh_copy.prim, mesh_in_robot)
 
+                if self.mode == "base":
+                    mesh_copy.collision_enabled = False
+                elif self.mode == "arm":
+                    mesh_copy.collision_enabled = True
 
+        if self.robot.model_name == "Tiago" and self.mode == "base":
+            og.sim.remove_object(robot_to_copy)
+
+        self._disable_robot_colliders()
+        og.sim.step()
+
+    def _set_prim_pose(self, prim, pose):
+        translation = Gf.Vec3d(*np.array(pose[0], dtype=float))
+        prim.GetAttribute("xformOp:translate").Set(translation)
+        orientation = np.array(pose[1], dtype=float)[[3, 0, 1, 2]]
+        prim.GetAttribute("xformOp:orient").Set(Gf.Quatd(*orientation)) 
```

Comment:
See later comment on one of the below files on this. If we have the prim as an OmniGibson object, there's no gain from calling this API manually